Pod/Classes/SEGAdobeIntegration.m

```diff
@@ -291,4 +316,53 @@ - (NSString *)formatProducts:(NSDictionary *)obj
     return [output componentsJoinedByString:@";"];
 }
 
+///-------------------------
+/// @name Video Tracking
+///-------------------------
+
+-(void)trackHeartbeatEvents:(SEGTrackPayload *)payload
+{
+    if ([payload.event isEqualToString:@"Video Playback Started"]){
+        self.config = [self createConfig:payload];
+       self.ADBMediaHeartbeat = [self initializeMediaHeartbeat:self.config andDelegate:nil];
+        [self.ADBMediaHeartbeat trackSessionStart:nil data:@{}];
+        return;
+    }
+}
+
+-(ADBMediaHeartbeatConfig *)createConfig:(SEGTrackPayload *)payload
+{
+    NSDictionary *properties = payload.properties;
+    NSDictionary *options = payload.integrations[@"Adobe Analytics"];
+    
+    self.config = [[ADBMediaHeartbeatConfig alloc] init];
+    bool sslEnabled = @NO;
+    if ([self.settings[@"ssl"] isEqual:@YES]) {
+        sslEnabled = @YES;
+    }
+    
+    bool debugEnabled = @NO;
+    if ([options[@"debug"] isEqual:@YES]) {
+        debugEnabled = @YES;
+    }
+    
+    NSMutableDictionary *infoDictionary = [[[NSBundle mainBundle] infoDictionary] mutableCopy];
+    [infoDictionary addEntriesFromDictionary:[[NSBundle mainBundle] localizedInfoDictionary]];
+    self.config.appVersion = infoDictionary[@"CFBundleShortVersionString"] ?: @"";
+    
+    self.config.trackingServer = self.settings[@"heartbeatTrackingServer"];
+    self.config.channel = properties[@"channel"];
+    self.config.ovp = options[@"ovp_name"];
+    self.config.playerName = properties[@"video_player"];
+    self.config.ssl = sslEnabled;
+    self.config.debugLogging = debugEnabled;
+    
+    return self.config;
+}
+
+-(ADBMediaHeartbeat*)initializeMediaHeartbeat:(ADBMediaHeartbeatConfig *)config andDelegate:(id<ADBMediaHeartbeatDelegate>)playerDelegate
```

Comment:
does this need to be a method? it's only called from one place, and I think you could just do `[self.ADBMediaHeartbeatFactory createWithDelegate:playerDelegate andConfig:config]` there.

Reply:
sounds good, was just trying to abstract as much as possible to simplify